satpy/tests/test_writers.py

```diff
@@ -118,41 +118,41 @@ def test_init_nonexistent_enh_file(self):
             Enhancer(enhancement_config_file="is_not_a_valid_filename_?.yaml")
 
 
+class _CustomImageWriter(ImageWriter):
+    def __init__(self, **kwargs):
+        super().__init__(name="test", config_files=[], **kwargs)
+        self.img = None
+
+    def save_image(self, img, **kwargs):
+        self.img = img
+
+
 class _BaseCustomEnhancementConfigTests:
 
     TEST_CONFIGS: dict[str, str] = {}
 
-    @classmethod
-    def setup_class(cls):
-        """Create fake user configurations."""
-        for fn, content in cls.TEST_CONFIGS.items():
-            base_dir = os.path.dirname(fn)
-            if base_dir:
-                os.makedirs(base_dir, exist_ok=True)
-            with open(fn, "w") as f:
-                f.write(content)
+    @pytest.fixture(scope="class", autouse=True)
+    def test_configs_path(self, tmp_path_factory):
+        """Create test enhancement configuration files in a temporary directory.
 
-        # create fake test image writer
-        from satpy.writers import ImageWriter
+        The root temporary directory is changed to and returned.
 
-        class CustomImageWriter(ImageWriter):
-            def __init__(self, **kwargs):
-                super(CustomImageWriter, self).__init__(name="test", config_files=[], **kwargs)
-                self.img = None
+        """
+        prev_cwd = pathlib.Path.cwd()
+        tmp_path = tmp_path_factory.mktemp("config")
+        os.chdir(tmp_path)
```

Comment:
What is the `chdir` needed for?

Reply:
The tests are testing that configs in the current working directory are discovered and used. However, now that I say that that might not be important to all the tests. I'll try to review the tests and see if this chdir can be limited to only a few tests.

Reply:
So there is at least one class of tests that I think it is important that the configs come from the current working directory. Or at least it is nice that they test it. The other tests would be fine if the config path was instead added using the `satpy.config` interface(s) for adding user custom configs. Either way, this current directory versus satpy.config directory stuff should be handled by other `satpy.config`-specific tests and is not necessary to test here. The tests look about the same whether `satpy.config` is used versus `chdir`, but I think the `chdir` ends up being a little cleaner but this is more of a personal preference and if I'm being honest some level of laziness that I don't want to change it.

Unless someone feels strongly I think I'd like this merged as-is and if more cleanup is needed in the future I can do that.